### PR TITLE
test: Simplify Jest config and expand coverage

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

Our Mocha and Jest configuration has been messy because we're in the middle of a migration from Mocha to Jest. Each file had to be explicitly included in the Jest configuration or ignored in the Mocha configuration. This was inconvenient and error-prone, and resulted in some tests not being run at all.

~Both test configurations have been updated to use a shared list of Mocha test files. There are only a few of these files left, and this list should only get shorter as we migrate more tests to Jest. No further configuration changes will be needed to add Jest tests.~

We have now finished migrating unit tests from Mocha to Jest, so this PR now only affects the Jest configuration.

Note that the ESLint configuration has not been updated to use these simpler globs to determine which tests use Mocha and which use Jest. I tried doing that in this PR initially but it raised too many lint errors, so that will come in a later PR. In the meantime, we may still need to update `.eslintrc.js` when adding a new test of either type.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25013?quickstart=1)

## **Related issues**

N/A

## **Manual testing steps**

Run `yarn test:unit` and verify that it runs correctly (no errors, no missing tests)

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Gudahtt** [comment]
This is blocked by these two PRs:
* https://github.com/MetaMask/metamask-extension/pull/25012
* https://github.com/MetaMask/metamask-extension/pull/25014

**Comment by Gudahtt** [comment]
Moving into draft until lint errors are resolved

**Comment by Gudahtt** [comment]
Rebased to resolve conflicts, and the Mocha test coverage list has been removed because it was now empty.

**Comment by Gudahtt** [comment]
More conflicts to resolve. Moving into draft (CI passes but it shouldn't, this is reducing test coverage).

File: jest.config.js [hunk 1]
```diff
@@ -1,26 +1,8 @@
...
-    '<rootDir>/app/scripts/controllers/transactions/IncomingTransactionHelper.ts',
-    '<rootDir>/app/scripts/controllers/preferences.js',
-    '<rootDir>/app/scripts/flask/**/*.js',
-    '<rootDir>/app/scripts/lib/**/*.(js|ts)',
-    '<rootDir>/app/scripts/metamask-controller.js',
-    '<rootDir>/app/scripts/migrations/*.js',
-    '<rootDir>/app/scripts/migrations/*.ts',
-    '!<rootDir>/app/scripts/migrations/*.test.(js|ts)',
-    '<rootDir>/app/scripts/platforms/*.js',
+    '<rootDir>/app/scripts/**/*.(js|ts|tsx)',
```
Comment (vthomas13) [inline jest.config.js:3]: wow, nice

**Review by vthomas13 — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**